packages/vite/src/node/server/http.ts

```diff
@@ -44,9 +45,19 @@ export async function resolveHttpsConfig(
     key: readFileIfExists(key),
     pfx: readFileIfExists(pfx)
   })
-  if (!httpsOption.key || !httpsOption.cert) {
+
+  if (!httpsOption.pfx && (!httpsOption.key || !httpsOption.cert)) {
     httpsOption.cert = httpsOption.key = await getCertificate(config)
   }
+
+  if (httpsOption.pfx && httpsOption.passphrase) {
+    const buf = Buffer.from(new String(httpsOption.passphrase))
+    // type definition is incorrect; Buffer should be allowed per Node docs
```

Comment:
In which part of the docs?

I only find that it accepts a `string` type: https://nodejs.org/docs/latest-v16.x/api/tls.html#:~:text=passphrase%20%3Cstring%3E%20Shared%20passphrase%20used%20for%20a%20single%20private%20key%20and/or%20a%20PFX.

Reply:
I apologize, I may have misread it.  The problem is that when I use my sample repo and make the change to vite to at least not generate a cert/key when a PFX is specified, I get this error:

```
$ vite --https.pfx .\localhost.pfx --https.passphrase 1234
error when starting dev server:
TypeError: Pass phrase must be a buffer
    at Object.createSecureContext (_tls_common.js:271:19)
    at Http2SecureServer.Server.setSecureContext (_tls_wrap.js:1321:27)
    at new Server (_tls_wrap.js:1185:8)
    at new Http2SecureServer (internal/http2/core.js:2870:5)
    at Object.createSecureServer (internal/http2/core.js:3037:10)
    at resolveHttpServer (C:\Data\Git\vite\packages\vite\src\node\server\http.ts:24:29)
    at createServer (C:\Data\Git\vite\packages\vite\src\node\server\index.ts:317:13)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at CAC.<anonymous> (C:\Data\Git\vite\packages\vite\src\node\cli.ts:84:22)
error Command failed with exit code 1.
```

I'm not sure what else to do.

Reply:
Here's the location of the function that creates the context in Node source:
https://github.com/nodejs/node/blob/fe3c5a769d6a3835d757718804c5d159ca1981c7/src/crypto/crypto_context.cc#L923

And here's the actual throwing of the error.  It seems to require Buffer for PFX.
https://github.com/nodejs/node/blob/fe3c5a769d6a3835d757718804c5d159ca1981c7/src/crypto/crypto_context.cc#L943

```
 if (args.Length() >= 2) {
    THROW_AND_RETURN_IF_NOT_BUFFER(env, args[1], "Pass phrase");
    Local<ArrayBufferView> abv = args[1].As<ArrayBufferView>();
    size_t passlen = abv->ByteLength();
    pass.resize(passlen + 1);
    abv->CopyContents(pass.data(), passlen);
    pass[passlen] = '\0';
  }
```